src/borg/archive.py

```diff
@@ -1034,28 +1034,35 @@ def identify_key(self, repository):
         return key_factory(repository, cdata)
 
     def verify_data(self):
+        def chunker(it, size):
+            while True:
+                result = []
+                try:
+                    for i in range(size):
+                        result.append(next(it))
+                except StopIteration:
+                    yield result
+                    raise StopIteration
+                else:
+                    yield result
+
         logger.info('Starting cryptographic data integrity verification...')
         count = len(self.chunks)
         errors = 0
         defect_chunks = []
         pi = ProgressIndicatorPercent(total=count, msg="Verifying data %6.2f%%", step=0.01)
-        for chunk_id, (refcount, *_) in self.chunks.iteritems():
-            pi.show()
-            try:
-                encrypted_data = self.repository.get(chunk_id)
-            except Repository.ObjectNotFound:
-                self.error_found = True
-                errors += 1
-                logger.error('chunk %s not found', bin_to_hex(chunk_id))
-                continue
```

Comment:
hmm, guess we have - esp. in borg check.

if a segment file is lost or truncated or ... a object in repo index might be not in segment file any more.